keps/sig-node/2837-pod-level-resource-spec/README.md

```diff
@@ -0,0 +1,1950 @@
+in-place pod resize for pod-level resource allocation, users should be able to
+either modify the pod-level resources to accommodate ephemeral containers or
+supply resources at container-level for ephemeral containers and kubernetes will
+resize the pod to accommodate the ephemeral containers.
+
+#### [Scoped for Beta] VPA
+
+TBD. Do not review for the alpha stage.
+
+#### [Scoped for Beta] Cluster Autoscaler
+
+Cluster Autoscaler won't work as expected with pod-level resources in alpha since
+it relies on container-level values to be specified. If a user specifies only
+pod-level resources, the CA will assume that the pod requires no resources since
+container-level values are not set. As a result, the CA won't scale the number of
+nodes to accommodate this pod. Meanwhile, the scheduler will evaluate the
+pod-level resource requests but may be unable to find a suitable node to fit the
+pod. Consequently, the pod will not be scheduled. While this behavior is
+acceptable for the alpha implementation, it is anticipated that Cluster
+Autoscaler support will be addressed in the Beta phase with pod resource
+requirements surfaced in a helper library/function that autoscalers can use to
+make autoscaling decisions.
+
+#### [Scoped for Beta] Support for Windows
+
+Pod-level resource specifications are a natural extension of Kubernetes' existing
+resource management model. Although this new feature is expected to function with
+Windows containers, careful testing and consideration are required due to
+platform-specific differences. As the introduction of pod-level resources is a
+major change in itself, full support for Windows will be addressed in future
+stages, beyond the initial alpha release.
+
+### Test Plan
+
+[X] I/we understand the owners of the involved components may require updates to
+existing tests to make this code solid enough prior to committing the changes
+necessary to implement this enhancement.
+
+#### Unit tests
+
+This feature will touch multiple components. For alpha, unit tests coverage for following packages needs to be added:
+
+* Scheduler logic will be updated to consider pod-level requests. Hence pkg/scheduler will require additional coverage.
+* pkg/kubelet/cm will be updated to set pod-level cgroups using CPU requests and limits, and memory limits.
+* pkg/apis/core/validation/types_test.go and pkg/apis/core/validation since new fields are added in PodSpec and also new validation rules are required for the new fields.
+* pkg/kubeapiserver/admission for changes made in the admission logic for LimitRanger and ResourceQuota admission controllers.
+
+
+#### e2e tests
+
+Following scenarios need to be covered:
+
+
+
+* Cgroup settings when pod-level resources are set.
+* Validate scheduling and admission.
+* Validate the containers with no limits set are throttled on CPU when CPU usage reaches Pod level CPU limits.
+* Validate the containers with no limits set are OOMKilled when memory usage reaches Pod level memory limits.
+
```

Comment:
Test the correct values in `TotalResourcesRequested`

Reply:
Adding it, although TotalResourcesRequested will be added in beta.